ecl/hql/hqlpmap.cpp

```diff
@@ -524,8 +524,15 @@ IHqlExpression * NewProjectMapper2::recursiveExpandSelect(IHqlExpression * expr,
         unsigned match = targets.find(*plain);
         if (match == NotFound)
         {
-            if (ignoreMissingFields)
-                return LINK(expr);
+            /*
+             * Sometimes the implicit project code will remove fields that are used in WILD() expressions.  This creates an inconsistency
+             * in the graph - a reference to a field that doesn't exist in the output of the previous expression.  If we are expanding
+             * a selector reference, and it isn't found in the input's dataset, assume it has been projected out, and remap it to the
+             * parent dataset.  (If this is wrong it will trigger an internal error later on.)
```

Comment:
I have none other than to look further. One of my concerns is whether it is possible that future code changes will require knowledge of this PR to prevent creating a scenario that isn't caught because it relies on other assertions rather than this code change having it's own. This is ok if the other assertions are core/base/foundation assertions but if they are not... It's a question of containment.